plugins/node/opentelemetry-instrumentation-connect/src/instrumentation.ts

```diff
@@ -120,16 +120,16 @@ export class ConnectInstrumentation extends InstrumentationBase<Server> {
       if (!instrumentation.isEnabled()) {
         return (middleWare as any).apply(this, arguments);
       }
-      const [reqArgIdx, resArgIdx, nextArgIdx] = isErrorMiddleware
-        ? [1, 2, 3]
-        : [0, 1, 2];
-      const req = arguments[reqArgIdx] as IncomingMessage;
+      const [resArgIdx, nextArgIdx] = isErrorMiddleware ? [2, 3] : [1, 2];
       const res = arguments[resArgIdx] as ServerResponse;
       const next = arguments[nextArgIdx] as NextFunction;
 
       const rpcMetadata = getRPCMetadata(context.active());
       if (routeName && rpcMetadata?.type === RPCType.HTTP) {
-        rpcMetadata.span.updateName(`${req.method} ${routeName || '/'}`);
+        setRPCMetadata(
+          context.active(),
+          Object.assign(rpcMetadata, { route: routeName || '/' })
```

Comment:
Why not simply set the route instead of using `Object.assign`? e.g.
```ts
if (routeName && rpcMetadata?.type === RPCType.HTTP) {
  rpcMetadata.route = routeName;
}
```
There should be also no need to default to `/` as the if ensures that it is no empty.

Reply:
update, I follow this pattern from some of the other frameworks.